src/zon.zig

```diff
@@ -0,0 +1,831 @@
+                    .f80_type => .{ .f80 = @floatFromInt(val) },
+                    .f128_type, .comptime_float_type => .{ .f128 = @floatFromInt(val) },
+                    else => unreachable,
+                },
+            } }),
+            .big => |val| return self.sema.pt.intern(.{ .float = .{
+                .ty = res_ty.toIntern(),
+                .storage = switch (res_ty.toIntern()) {
+                    .f16_type => .{ .f16 = val.toFloat(f16) },
+                    .f32_type => .{ .f32 = val.toFloat(f32) },
+                    .f64_type => .{ .f64 = val.toFloat(f64) },
+                    .f80_type => .{ .f80 = val.toFloat(f80) },
+                    .f128_type, .comptime_float_type => .{ .f128 = val.toFloat(f128) },
+                    else => unreachable,
+                },
+            } }),
+        },
+        .float_literal => |val| return self.sema.pt.intern(.{ .float = .{
+            .ty = res_ty.toIntern(),
+            .storage = switch (res_ty.toIntern()) {
+                .f16_type => .{ .f16 = @floatCast(val) },
+                .f32_type => .{ .f32 = @floatCast(val) },
+                .f64_type => .{ .f64 = @floatCast(val) },
+                .f80_type => .{ .f80 = @floatCast(val) },
+                .f128_type, .comptime_float_type => .{ .f128 = val },
+                else => unreachable,
+            },
+        } }),
+        .pos_inf => return self.sema.pt.intern(.{ .float = .{
+            .ty = res_ty.toIntern(),
+            .storage = switch (res_ty.toIntern()) {
+                .f16_type => .{ .f16 = std.math.inf(f16) },
+                .f32_type => .{ .f32 = std.math.inf(f32) },
+                .f64_type => .{ .f64 = std.math.inf(f64) },
+                .f80_type => .{ .f80 = std.math.inf(f80) },
+                .f128_type, .comptime_float_type => .{ .f128 = std.math.inf(f128) },
+                else => unreachable,
+            },
+        } }),
+        .neg_inf => return self.sema.pt.intern(.{ .float = .{
+            .ty = res_ty.toIntern(),
+            .storage = switch (res_ty.toIntern()) {
+                .f16_type => .{ .f16 = -std.math.inf(f16) },
+                .f32_type => .{ .f32 = -std.math.inf(f32) },
+                .f64_type => .{ .f64 = -std.math.inf(f64) },
+                .f80_type => .{ .f80 = -std.math.inf(f80) },
+                .f128_type, .comptime_float_type => .{ .f128 = -std.math.inf(f128) },
+                else => unreachable,
+            },
+        } }),
+        .nan => return self.sema.pt.intern(.{ .float = .{
+            .ty = res_ty.toIntern(),
+            .storage = switch (res_ty.toIntern()) {
+                .f16_type => .{ .f16 = std.math.nan(f16) },
+                .f32_type => .{ .f32 = std.math.nan(f32) },
+                .f64_type => .{ .f64 = std.math.nan(f64) },
+                .f80_type => .{ .f80 = std.math.nan(f80) },
+                .f128_type, .comptime_float_type => .{ .f128 = std.math.nan(f128) },
```

Comment:
`comptime_float` is a little different from normal `f128` in that it can't be NaN (I don't recall if it can be inf, I think not). So, the comptime_float case here should be a compile error.